### PR TITLE
Extract ElfRelocatableSection.

### PR DESCRIPTION
**Pull request by corona10**
* Extract Sym.name from of ElfRelocatableSection.

Signed-off-by: Dong-hee Na donghee.na92@gmail.com

### CODE REVIEW

File: src/compiledcode/elf.py [hunk 1]
```diff
@@ -715,4 +716,20 @@ class ElfRelocatablesSection(object):
...
+        
+    def parse(self, elf, file_like):
+        while 1:
+            line = next_line(file_like)
+            if not line or re.match(self.end_re, line):
+                break
+            line = line.strip()
+            match = RELOC_RE().match(line)
+            if match:
+                sym = match.groups()[4]
```
Comment (corona10) [inline src/compiledcode/elf.py:735]: @pombredanne 
Should I distinguished with ''.rel.dyn' and  ''.rel.plt' or more?

**Comment by corona10** [comment]
@pombredanne 
For the reviewer, I want to export this data into like [this](https://github.com/nexB/scancode-toolkit-contrib/blob/develop/tests/compiledcode/data/elf/libelf.so.expected_elf_symbols).

But I have no idea which codes are related to this feature.
So please give me some hint to do with.
I think this practice will help me to analysis to a proposal for GSoC.

**Comment by pombredanne** [comment]
Thanks. I will post comment.

File: src/compiledcode/elf.py (hunk 1 above)
Comment (pombredanne) [reply src/compiledcode/elf.py:735]: not needed IMHO: when would this matter in our use case?

File: src/compiledcode/elf.py [hunk 2]
```diff
@@ -652,6 +651,8 @@ class ElfDebugPubnamesSection(object):
     """
     pass
 
+def RELOC_RE():
+    return re.compile(r"^([0-9a-fA-F]+)\s+([A-Fa-f0-9]+)\s+(\w+)\s+([A-Fa-f0-9]+)\s+(\w+)")
```
Comment (pombredanne) [inline src/compiledcode/elf.py]: a small comment explaining what a regex does is often useful

**Comment by pombredanne** [comment]
>But I have no idea which codes are related to this feature. So please give me some hint to do with.

For this test, JSON was written an as expectation on disk and then used like here: https://github.com/nexB/scancode-toolkit-contrib/blob/98d48859b9f9dccd6ef585c19e6cb35dadc1717e/tests/compiledcode/test_elf.py#L225

But this would be best refactored in a check test helper such as used for whole scans checks in https://github.com/nexB/scancode-toolkit/blob/16685bd94a7dd091f53ccfae8ec6f7d30f8c15bc/src/scancode/cli_test_utils.py#L67 or in https://github.com/nexB/scancode-toolkit/blob/develop/tests/packagedcode/packages_test_utils.py#L54

Actually there should likely be a common tests helper for these all. the point is that you can create a test helper function that accepts the path of the file to check and the path to a JSON file containing expected results. if a `regen` flag is True, you overwrite the expected file with dump of the current results. 
Then you run you function and compare the reloaded expected JSON with the results of the function run.

**Comment by pombredanne** [comment]
Looking good... but it is weird that it makes other parts of the code failing. Can you look into the tests failures?

**Comment by pombredanne** [comment]
I think the tests failures are not related. Let me review and merge

**Comment by corona10** [comment]
@pombredanne Sorry for a late response. I saw the PR which solved unit tests. That looks great 👍

**Comment by pombredanne** [comment]
Mreging. Thanks+

**Pull request merged**